src/OpenLoco/Objects/ObjectManager.cpp

```diff
@@ -816,34 +895,54 @@ namespace OpenLoco::ObjectManager
         call(0x00472031);
     }
 
-    void unload(LoadedObjectIndex index)
+    void unload(const LoadedObjectHandle& handle)
     {
-        callObjectFunction(index, ObjectProcedure::unload);
+        callObjectFunction(handle, ObjectProcedure::unload);
     }
 
-    size_t getByteLength(LoadedObjectIndex id)
+    size_t getByteLength(const LoadedObjectHandle& handle)
     {
-        return objectEntries[id].dataSize;
+        return objectEntries[getTypeOffset(handle.type) + handle.id].dataSize;
     }
 
-    std::vector<ObjectHeader> getHeaders()
+    template<typename TObject>
+    static void addAllInUseHeadersOfType(std::vector<ObjectHeader>& entries)
     {
-        std::vector<ObjectHeader> entries;
-        entries.reserve(ObjectManager::maxObjects);
-
-        for (size_t i = 0; i < ObjectManager::maxObjects; i++)
+        if constexpr (getMaxObjects(TObject::_objectType) == 1)
         {
-            auto obj = ObjectManager::get<Object>(i);
+            auto* obj = ObjectManager::get<TObject>();
             if (obj != nullptr)
             {
-                auto entry = getHeader(i);
+                auto entry = getHeader(getTypeOffset(TObject::_objectType));
                 entries.push_back(*entry);
             }
-            else
+        }
```

Comment:
Is there still an advantage to branching depending one or more objects here? Just having the for loop seems sensible at this point.

Reply:
Its due to having different functions for single objects and for multi objects.

Reply:
Oh right, the `ObjectManager::get` function does not take a parameter for single objects. I see.